Text/Internal/Css.hs

```diff
@@ -36,61 +38,60 @@ type CssUrl url = (url -> [(T.Text, T.Text)] -> T.Text) -> Css
 
 type DList a = [a] -> [a]
 
--- FIXME great use case for data kinds
-data Resolved
-data Unresolved
+data Resolved = Resolved | Unresolved
 
-type family Selector a
-type instance Selector Resolved = Builder
-type instance Selector Unresolved = [Contents]
-
-type family ChildBlocks a
-type instance ChildBlocks Resolved = ()
-type instance ChildBlocks Unresolved = [(HasLeadingSpace, Block Unresolved)]
+-- Should mixins keep order (new version) or not (deprecated version)
+data Order = Ordered | Unordered deriving (Lift)
 
 type HasLeadingSpace = Bool
 
-type family Str a
-type instance Str Resolved = Builder
-type instance Str Unresolved = Contents
-
-type family Mixins a
-type instance Mixins Resolved = ()
-type instance Mixins Unresolved = [Deref]
-
-data Block a = Block
-    { blockSelector :: !(Selector a)
-    , blockAttrs :: ![Attr a]
-    , blockBlocks :: !(ChildBlocks a)
-    , blockMixins :: !(Mixins a)
-    }
+type family Str (a :: Resolved)
+type instance Str 'Resolved = Builder
+type instance Str 'Unresolved = Contents
+
+data Block (a :: Resolved) where
+  BlockResolved :: 
+    { blockResSelector :: !Builder
+    , blockResAttrs    :: ![Attr 'Resolved]
+    } -> Block 'Resolved
+  BlockUnresolved ::
+    { blockUnresSelector :: ![Contents]
+    , blockUnresAttrs    :: ![Either (Attr 'Unresolved) Deref]
+    , blockUnresBlocks   :: ![(HasLeadingSpace, Block 'Unresolved)]
+    , blockUnresMixins   :: ![Deref]
```

Comment:
I think we can get rid of this field - in the "unordered case", we can just put the mixin `Deref`s at the end of `blockUnresAttrs`. I think this would simplify the code a bit.

(It's possible that we also would want to extend this consistent ordering to subblocks, but I think that's beyond the scope of this PR.)

Reply:
:+1:  I've got rid of the `blockUnresMixins`. Now we collect all mixins in the `buAttrsAndMixins` list, as you suggested.